File: 01-Creational/1.1 Abstract Factory/csharp/Bumper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AbstractFactory
{
    public abstract class Bumper
    {
        int totalNumberofParts;
        public Bumper(int numberofParts) {
            this.totalNumberofParts = numberofParts;
        }
        public void startAssemblyLine() {
            Console.WriteLine("Starting assembly line for Bumpers...");
        }
        public void fillBatch() {
            for(int i = 1; i <= totalNumberofParts; i++)
            {
                Console.WriteLine("part " + i + " finished...");
                Console.WriteLine("part " + i + " painted...");
                Console.WriteLine("part " + i + " stored...");
            }
        }
        public void cleanAssemblyLine(){
            Console.WriteLine("Cleaning working area...");
        }
        public  abstract void bringPiecesFromInventory();
    }
}

File: 01-Creational/1.1 Abstract Factory/Java/Bumper.java
package com.yeyocastillo.AbstractFactory;

public abstract class Bumper {
	Integer totalNumberofParts;
	
	Bumper(Integer numberofParts){
		this.totalNumberofParts = numberofParts;
	}

	void startAssemblyLine(){
		System.out.println("Starting Assembly Line..");
	}
	
	void fillBatch(){
		for (int i =1; i <= totalNumberofParts; i++)
		{
			System.out.println("part "+ i +" finished..");
			System.out.println("part "+ i +" painted..");
			System.out.println("part "+ i +" stored..");
		}
	}
	void cleanAssemblyLine(){
		System.out.println("Cleaning working area...");
	}
	protected abstract void bringPiecesFromInventory();
}
